azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+											Schema: map[string]*schema.Schema{
+												"direction": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ScaleDirectionDecrease),
+														string(insights.ScaleDirectionIncrease),
+													}, false),
+												},
+												"type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ChangeCount),
+														string(insights.PercentChangeCount),
+													}, false),
+												},
+												"value": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"cooldown": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: iso8601DurationString(),
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+								},
+							},
+						},
+						"fixed_date": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MinItems:      1,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.recurrence"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"time_zone": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+									},
+									"start": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: rfc3339String(),
+									},
+									"end": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: rfc3339String(),
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+						"recurrence": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MinItems:      1,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.fixed_date"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"frequency": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(insights.Week),
+										}, false),
+									},
+									"schedule": {
+										Type:     schema.TypeSet,
+										Required: true,
+										MinItems: 1,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"time_zone": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+												},
+												"days": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
```

Comment:
Passing values like "sunday" will not result in any REST API errors, however, the web portal UI will not show "Sunday" as checked (sigh). And by custom validation method do you mean custom logic when reading the values from configuration? Because the adding the ValidateFunc here will result in runtime error: ValidateFunc is not yet supported on lists or sets.